plugins/nodes/src/js/components/NodesTable.tsx

```diff
@@ -182,6 +191,16 @@ export default class NodesTable extends React.Component<
             cellRenderer={healthRenderer}
           />
 
+          <Column
+            header={
+              <HeaderCell>
```

Comment:
Why are we making this non-sortable?

Reply:
Yeah, this should be sortable please

Reply:
Oh sorry i was under the Impression that we should take the route which would take the least effort. Making this sortable means that we need to take another approach to this problem. Right now we can’t make it sortable since we only now the ip‘s of the current visible nodes. Also the nodes information is not getting merged with the nodes network information. Should we implement this sortable also if it means more effort since we need to merge the nodes information at a higher level?

Reply:
I disagree here and in retrospect I understand where we had a miscomminucation, sorry for not clearing that up earlier. I believe that the current solution is not good enough, we should not compromise on a product level.

So what we currently do is that every Cell asks for their data, so the overall architecture of this table "knowing about everything in advance" is violated. This leads to the sorting not working properly and also to the Autosizer jamming up.

What we also could do without changing and complicating the data layer implementations and what I thought was the plan to begin with is that we request all data beforehand. As we don't want to scope creep and move the entire table to the data layer I would suggest an approach like this:


```js
const queryDataLayerForNodeAndReturnPublicIp = (hostname) => {};
// Put in the same props as you would in the NodesTable
export default const NodesTableWrapper = componentFromStream(props$ => {
  const data$ = props$.map(props => props.data);
  const dataWithPublicIp$ = props$.map(data => {
     const streamArray = data.map(item => {
       const publicIp$ = of(item).map(item => 
           queryDataLayerForNodeAndReturnPublicIp(item.hostname);
       );

       return item$.combineLatest(publicIp$, (item, publicIp) => 
          ({...item, publicIp })
       );
     });

     return of(42).combineLatest(...streamArray, (_ignore, ...items => items);

  return data$.combineLatest(dataWithPublicIp$, (props, data) => ({...props, data})).map(props => <NodesTable {...props} />)
});
```

I know this example is a bit lengthy, but I think we should take that route or another that leads to us shipping a feature without compromise

Reply:
To be clear, I would like to block this for that reason, but I'm OOO for two weeks now

Reply:
I did my best to add the example of @DanielMSchmidt  which was unfortunately not as easy as laid out. I choose to select the lowest level possible so that I only have to interact with Node structs and not Nodeslist.